data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsPublisher.java

```diff
@@ -38,10 +38,11 @@ public int publishMetrics(final String endpointAddress, final String json) throw
     RequestBody body = RequestBody.create(json, mediaType);
     Request request = new Request.Builder().url(endpoint).post(body).build();
     Response response = this.client.newCall(request).execute();
-    if (response.code() != 200) {
-      LOG.error(
-          "Failed to publish metrics to {}. Response code {}", endpointAddress, response.code());
+    int responseCode = response.code();
+    response.close();
+    if (responseCode != 200) {
```

Comment:
not sure we want a log.error every time the receiver is not available - i'd be tempted to make this a log.debug otherwise we may well flood our logs very quickly, might check with @ajsutton on this

Reply:
hmm, might live to regret this but I think error is the right level here since nothing else can really report the failure. It helps that its only a single line and not a full stack trace.

One possible concern - could `endpointAddress` contain a secret token as part of the URL? I think I'd be tempted to omit it and just report `Failed to publish metrics to external metrics service. Response code {}`

Reply:
yes, good point, the endpoint could contain a secret token. I'll remove the parameter from the log.